pipert2/utils/method_data.py

```diff
@@ -1,13 +1,25 @@
+import json
 from typing import Dict, Optional
 from collections import defaultdict
 from pipert2 import dataclass, field
 
 
 @dataclass
 class Method:
-    event_name: str
-    specific_flow_routines: Dict[str, Optional[list]] = field(default_factory=lambda: defaultdict(list))
-    params: dict = field(default_factory=lambda: {})
+    def __init__(self, event_name, specific_flow_routines=None, params=None):
+        self.event_name = event_name
+
+        if specific_flow_routines is None:
+            self.specific_flow_routines = defaultdict(list)
+        elif isinstance(specific_flow_routines, str):
+            self.specific_flow_routines = json.loads(specific_flow_routines)
+        else:
+            self.specific_flow_routines = specific_flow_routines
+
+        if params is None:
+            self.params = {}
+        else:
+            self.params = params
```

Comment:
IMO, the init function and what was before are the same except for string sent to `specific_flow_routines` and then u convert it to dict from json loads.
I don't think you should change things here and have the json loads when you create the Method when you know that the `specific_flow_routines`  is actually a string.

Reply:
You right